engine/src/main/java/com/datatorrent/stram/StramClient.java

```diff
@@ -445,7 +445,8 @@ public void startApplication() throws YarnException, IOException
 
     // copy required jar files to dfs, to be localized for containers
     try (FileSystem fs = StramClientUtils.newFileSystemInstance(conf)) {
-      Path appsBasePath = new Path(StramClientUtils.getDTDFSRootDir(fs, conf), StramClientUtils.SUBDIR_APPS);
+      Path appsBasePath = new Path(StramClientUtils.getApexDFSRootDir(fs, conf,
```

Comment:
Get the getApexDFSRootDir figure out how to get the current user, it is already doing that for the login user.